src/Microsoft.AspNetCore.Mvc.Razor/Compilation/MetadataReferenceFeatureProvider.cs

```diff
@@ -0,0 +1,80 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Reflection.PortableExecutable;
+using Microsoft.AspNetCore.Mvc.ApplicationParts;
+using Microsoft.CodeAnalysis;
+using Microsoft.Extensions.DependencyModel;
+
+namespace Microsoft.AspNetCore.Mvc.Razor.Compilation
+{
+    /// <summary>
+    /// An <see cref="IApplicationFeatureProvider{TFeature}"/> for <see cref="MetadataReferenceFeature"/> that 
+    /// uses <see cref="DependencyContext"/> for registered <see cref="AssemblyPart"/> instances to create 
+    /// <see cref="MetadataReference"/>.
+    /// </summary>
+    public class MetadataReferenceFeatureProvider : IApplicationFeatureProvider<MetadataReferenceFeature>
+    {
+        /// <inheritdoc />
+        public void PopulateFeature(IEnumerable<ApplicationPart> parts, MetadataReferenceFeature feature)
+        {
+            if (parts == null)
+            {
+                throw new ArgumentNullException(nameof(parts));
+            }
+
+            if (feature == null)
+            {
+                throw new ArgumentNullException(nameof(feature));
+            }
+
+            var libraryPaths = new HashSet<string>(StringComparer.OrdinalIgnoreCase);
```

Comment:
It would rely on `DependencyContext` consistently casing this (including binaries that are compiled on different machines). Probably safer to leave it as is. Original code is here - https://github.com/aspnet/Mvc/pull/4672/files/549d113b37a6baed6871dee4be7721da0df6301e#diff-96cb3f263eeedd62fed8e95c300bb8b0L253